skimage/filters/tests/test_fft_based.py

```diff
@@ -0,0 +1,250 @@
+from skimage.filters import butterworth
+
+
+def _fft_centered(x):
+    return fftmodule.fftshift(fftmodule.fftn(fftmodule.fftshift(x)))
+
+
+def test_butterworth_2D_zeros():
+    im = np.zeros((4, 4))
+    filtered = butterworth(im)
+    assert filtered.shape == im.shape
+    assert_array_equal(im, filtered)
+
+
+@pytest.mark.parametrize("high_pass", [True, False])
+def test_butterworth_2D(high_pass):
+    # rough check of high-pass vs. low-pass behavior via relative energy
+    im = np.random.randn(64, 128)
+    filtered = butterworth(
+        im,
+        cutoff_frequency_ratio=0.20,
+        high_pass=high_pass,
+    )
+
+    # Compute the energy at the outer edges of the Fourier domain
+    # before and after filtering.
+    im_fft = _fft_centered(im)
+    im_fft = np.real(im_fft * np.conj(im_fft))
+    filtered_fft = _fft_centered(filtered)
+    filtered_fft = np.real(filtered_fft * np.conj(filtered_fft))
+    outer_mask = np.ones(im.shape, dtype=bool)
+    outer_mask[4:-4, 4:-4] = 0
+    abs_filt_outer = filtered_fft[outer_mask].mean()
+    abs_im_outer = im_fft[outer_mask].mean()
+
+    # Compute energy near the center of the Fourier domain
+    inner_sl = tuple(slice(s // 2 - 4, s // 2 + 4) for s in im.shape)
+    abs_filt_inner = filtered_fft[inner_sl].mean()
+    abs_im_inner = im_fft[inner_sl].mean()
+
+    if high_pass:
+        assert abs_filt_outer > 0.9 * abs_im_outer
+        assert abs_filt_inner < 0.1 * abs_im_inner
+    else:
+        assert abs_filt_outer < 0.1 * abs_im_outer
+        assert abs_filt_inner > 0.9 * abs_im_inner
+
+
+@pytest.mark.parametrize("high_pass", [True, False])
+def test_butterworth_2D_realfft(high_pass):
+    """Filtering a real-valued array is equivalent to filtering a
+       complex-valued array where the imaginary part is zero.
+    """
+    im = np.random.randn(32, 64)
+    kwargs = dict(
+        cutoff_frequency_ratio=0.20,
+        high_pass=high_pass
+    )
+    filtered_real = butterworth(im, **kwargs)
+    filtered_cplx = butterworth(im.astype(np.complex128), **kwargs)
+    assert_allclose(filtered_real, filtered_cplx.real)
+
+
+def test_butterworth_3D_zeros():
+    im = np.zeros((3, 4, 5))
+    filtered = butterworth(im)
+    assert filtered.shape == im.shape
+    assert_array_equal(im, filtered)
+
+
+def test_butterworth_4D_zeros():
+    im = np.zeros((3, 4, 5, 6))
+    filtered = butterworth(im)
+    assert filtered.shape == im.shape
+    assert_array_equal(im, filtered)
+
+
+@pytest.mark.parametrize(
+    "chan, dtype",
+    [(0, np.float64), (1, np.complex128), (2, np.uint8), (3, np.int64)],
+)
+def test_butterworth_4D_channel(chan, dtype):
+    im = np.zeros((3, 4, 5, 6), dtype=dtype)
+    filtered = butterworth(im, channel_axis=chan)
+    assert filtered.shape == im.shape
+
+
+def test_butterworth_correctness_bw():
+    small = coins()[180:190, 260:270]
+    filtered = butterworth(small,
+                           cutoff_frequency_ratio=0.2,
+                           )
```

Comment:
```suggestion
                           cutoff_frequency_ratio=0.2)
```
so it's consistent with line 121?

Reply:
done